react/ShippingHeader.tsx

```diff
@@ -36,7 +52,11 @@ const ShippingHeader: React.VFC<Props> = ({ onEditAddress }) => {
       </p>
       <div className="flex items-center mt2 mb6">
         <div className="lh-copy">
-          <PlaceDetails display="minimal" />
+          {data?.installedApp?.source === 'none' ? (
```

Comment:
add default value to avoid blinking

```suggestion
          {(data?.installedApp?.source ?? 'none') === 'none' ? (
```

you could also invert the condition, which I find is a bit more elegant

```ts
{data?.installedApp?.source !== 'none' ? (
  <PlaceDetails display="brief" />
) : (
  <PlaceDetails display="minimal" />
)}
```

Reply:
do you think it is worth to extract this logic into a "shared" app (like checkout-components)? since we are repeating this over and over across different apps, and whenever we need to change this we will need to open a PR in several repositories

Reply:
> you could also invert the condition, which I find is a bit more elegant

k

> since we are repeating this over and over across different apps

What do you mean? AFAIK it's only being used here. To be fair, it's being used in `NewAddressForm` and `ShippingHeader` so I guess we could at least create a hook for this logic (will push here soon)

Reply:
actually, before coding this, how should we handle this loading to avoid blinking?
as a query will be needed to decide which component to render:
1. we put a loading state, but that might look silly from an UX perspective, idk
2. we use LocationInput as a default, but if the loading takes too long, it can impact UX as well, bc the component will suddenly swap
3. we could preload this, but idk how
what do you think?

Reply:
we can use the `installedAppPublic` query, which is a bit faster than the `installedApp` (because it is publicly cacheable), and display a loading state while we wait for the query to execute

Reply:
noice will do